src/Microsoft.Azure.WebJobs.Host/InvocationFilterAttribute.cs

```diff
@@ -0,0 +1,39 @@
+﻿// InvocationFilterAttribute.cs
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// This is the first iteration for the Invocation Filter Attribute
+    /// </summary>
+    [CLSCompliant(false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = false)]
```

Comment:
I think you can remove this AttributeUsage annotation, since this type is abstract. Only the AttributeUsage declared on concrete subclasses will actually have any effect I believe. You should verify that.